fenix/app/src/main/java/org/mozilla/fenix/browser/tabstrip/TabStrip.kt

```diff
@@ -279,22 +284,37 @@ private fun TabItem(
             horizontalArrangement = Arrangement.SpaceBetween,
         ) {
             Row(
-                modifier = Modifier.weight(1f, fill = false),
+                modifier = Modifier.weight(1f),
                 verticalAlignment = Alignment.CenterVertically,
             ) {
+                // This makes sure that isRtl is only calculated when the title changes.
+                val isTitleRtl = remember(state.title) {
+                    BidiFormatter.getInstance().isRtl(state.title)
+                }
+
                 Spacer(modifier = Modifier.size(8.dp))
 
                 TabStripIcon(state.url)
 
                 Spacer(modifier = Modifier.size(8.dp))
 
-                Text(
-                    text = state.title,
-                    color = FirefoxTheme.colors.textPrimary,
-                    maxLines = 1,
-                    overflow = TextOverflow.Ellipsis,
-                    style = FirefoxTheme.typography.subtitle2,
-                )
+                HorizontalFadingEdgeBox(
+                    modifier = Modifier
+                        .weight(1f)
+                        .fillMaxHeight(),
+                    fadeWidth = longerTitleFadeWidth,
+                    backgroundColor = backgroundColor,
+                    isContentRtl = isTitleRtl,
+                ) {
+                    Text(
+                        text = state.title,
+                        color = FirefoxTheme.colors.textPrimary,
+                        maxLines = 1,
+                        softWrap = false,
+                        style = FirefoxTheme.typography.subtitle2,
+                        modifier = Modifier.align(Alignment.CenterStart),
```

Comment:
Could we reorder these parameters to follow the same order as they are defined https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/material/material/src/commonMain/kotlin/androidx/compose/material/Text.kt;l=93-110?

Reply:
Sure, I've updated. I want to mention that if we want to do this, we should write a lint rule, this way won't scale. Personally I don't see the value in this as named params solve the readibility problem.